shared/src/question-generators/StackQueue/QueueGenerator.ts

```diff
@@ -0,0 +1,818 @@
+import { Language } from "@shared/api/Language.ts"
+import {
+  minimalMultipleChoiceFeedback,
+  MultiFreeTextFeedbackFunction,
+  MultiFreeTextFormatFunction,
+  Question,
+  QuestionGenerator,
+} from "@shared/api/QuestionGenerator.ts"
+import { serializeGeneratorCall } from "@shared/api/QuestionRouter.ts"
+import { Queue } from "@shared/question-generators/StackQueue/Queue.ts"
+import { parseArrayString } from "@shared/question-generators/StackQueue/StackGenerator.ts"
+import Random from "@shared/utils/random.ts"
+import { t, tFunction, tFunctional, Translations } from "@shared/utils/translations.ts"
+
+const translations: Translations = {
+  en: {
+    name: "Queue-Implementation using an Array",
+    description: "Correctly use queue operations",
+    solutionFreetext: `|Index|Question|Solution|\n{{0}}`,
+    performOperations: `**We perform the following operations:**{{0}}`,
+    checkFormat: "Please only enter a number.",
+    checkFormatArray: "Please only enter numbers separated by commas.",
+    getQueueInfo:
+      "**Note:** The method **getQueue()** returns the complete queue as it is as a string. If the queue is not full, the remaining elements are filled with -1.",
+    toStringInfo:
+      "**Note:** The method **toString()** only returns the part between rear and front. Every other value from the field is ignored.",
```

Comment:
Die abstrakte Datenstruktur Queue hat keine Operationen `getQueueInfo` oder `toString`, wir sollten uns strikt an die abstrakte Datenstruktur halten (wie zum Beispiel in CLRS definiert). Ich würde diese Operationen vollständig streichen bzw. die Frage entsprechend anders formulieren.

Reply:
Ich verstehe was du meinst. Die Methoden `.getQueue()` und `.toString()` existieren eigentlich nicht. Ich würde gerne aber eine Frage haben, bei welcher der Nutzer angeben soll, wie das Array der Queue aussieht. 

Ich würde ungerne die Frage streichen, sondern lieber anders formulieren. Wäre `.getArray()` oder `getField()` besser? Dann würde ich `.toString()` streichen und `.getQueue()` umbenennen. 

So auch bei StackQuestion

Reply:
Wären dies bei der Queue dann nur die Operationen `enqueue` und `dequeue`?

Reply:
Auch `isEmpty`, aber sonst nichts, siehe Folie 6: https://files.tcs.uni-frankfurt.de/algo1/introtodatastructures.pdf
Wir müssen uns strikt an die Operationen der abstrakten Datenstruktur halten, das ist genau das Lernziel für diese Aufgabe, mit nur diesen Operationen umzugehen, sowie die interne Darstellung zu verstehen.

Die Frage "wie sieht das Array aus" ist sinnvoll, jedoch darf dies nicht über die Operationen der Datenstruktur geschehen. Stattdessen "Gib den Zustand der Datenstruktur an" mit Beispiel, wie die Eingabe formatiert sein soll.

Jetzt oder in einem zukünftigen pull request: Kannst du außerdem die Aufgaben 4.1a, 4.1c und 4.2 in genau der Formulierung von dem Übungsblatt als Variante implementieren? https://goethe-tcs.github.io/algo1-exercises/ALGO1-Blatt-04-introtodatastructures.pdf

Reply:
Okay, dann werde ich dies verändern, indem ich die Frage nach dem State des Feldes als weiteren Absatz einfüge. Also außerhalb der aktuellen Tabellenform.

Reply:
Die weiteren Aufgaben schreibe ich mir auf meine Todo-Liste und ergänze diese in einem zukünftigen PR. Dann wird es in diesem PR nicht noch größer.